packages/next/src/cli/next-dev.ts

```diff
@@ -265,18 +265,21 @@ const nextDev: CliCommand = async (argv) => {
       )
     }
 
-    const turboJson = findUp.sync('turbo.json', { cwd: dir })
     // eslint-disable-next-line no-shadow
     const packagePath = findUp.sync('package.json', { cwd: dir })
+    const lockfilePath = findUp.sync(
+      ['pnpm-lock.yaml', 'package-lock.json', 'yarn.lock'],
+      { cwd: dir }
+    )
```

Comment:
We could re-use the `rawNextConfig.experimental.outputFileTracingRoot` config as it already does this look x-ref: https://github.com/vercel/next.js/blob/56de0b2ba81591e7e24cb37d5499dafe281ec091/packages/next/src/server/config.ts#L480-L485